src/main/java/org/folio/rest/impl/PatronNoticePoliciesAPI.java

```diff
@@ -137,9 +158,17 @@ public void deletePatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePol
     Handler<AsyncResult<Response>> asyncResultHandler,
     Context vertxContext) {
 
-    PgUtil.deleteById(PATRON_NOTICE_POLICY_TABLE, patronNoticePolicyId, okapiHeaders,
-      vertxContext, DeletePatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse.class,
-      asyncResultHandler);
+    PostgresClient pgClient = PgUtil.postgresClient(vertxContext, okapiHeaders);
+
+    findCirculationRules(pgClient)
+      .map(CirculationRules::getRulesAsText)
+      .map(text -> text.contains(patronNoticePolicyId))
+      .compose(contains -> contains ? failedFuture(new NoticePolicyInUseException(IN_USE_POLICY_ERROR_MESSAGE)) : succeededFuture())
+      .compose(v -> deleteNoticePolicyById(pgClient, patronNoticePolicyId))
+      .map(v -> DeletePatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyIdResponse.respond204())
+      .map(Response.class::cast)
+      .otherwise(this::mapExceptionToResponse)
```

Comment:
I'm not a fan of managing flow of control (e.g. validation) with exceptions. However, I understand that we don't really have an alternative pattern in most modules

Reply:
I had doubts about this but such approach seemed to me the most convenient here.

Reply:
I think it is convenient, it seems to be the prevailing method in the storage modules